modules/currency-l0/src/test/scala/io/constellationnetwork/currency/l0/snapshot/SnapshotEventsPublisherDaemonSuite.scala

```diff
@@ -0,0 +1,122 @@
+package io.constellationnetwork.currency.l0.snapshot
+
+import java.security.KeyPair
+
+import cats.effect.std.{Queue, Supervisor}
+import cats.effect.{IO, Resource}
+
+import scala.concurrent.TimeoutException
+import scala.concurrent.duration.DurationInt
+
+import io.constellationnetwork.block.generators.signedBlockGen
+import io.constellationnetwork.currency.dataApplication.DataUpdate
+import io.constellationnetwork.currency.l0.snapshot.schema.{CurrencyConsensusKind, CurrencyConsensusOutcome}
+import io.constellationnetwork.currency.schema.currency.CurrencySnapshotContext
+import io.constellationnetwork.json.JsonSerializer
+import io.constellationnetwork.node.shared.config.types.{ConsensusConfig, EventCutterConfig}
+import io.constellationnetwork.node.shared.infrastructure.consensus.ConsensusStorage
+import io.constellationnetwork.node.shared.infrastructure.gossip.Gossip
+import io.constellationnetwork.node.shared.infrastructure.metrics.{Metrics, NoOpMetrics}
+import io.constellationnetwork.node.shared.infrastructure.snapshot.daemon.SnapshotEventsPublisherDaemon
+import io.constellationnetwork.node.shared.snapshot.currency.{BlockEvent, CurrencySnapshotArtifact, CurrencySnapshotEvent}
+import io.constellationnetwork.schema.generation.Generation
+import io.constellationnetwork.schema.gossip.{Counter, Ordinal, RumorRaw}
+import io.constellationnetwork.schema.peer.PeerId
+import io.constellationnetwork.security._
+import io.constellationnetwork.security.hex.Hex
+
+import eu.timepit.refined.types.numeric.{NonNegLong, PosInt, PosLong}
+import fs2.Stream
+import io.circe.{Encoder, Json}
+import weaver.MutableIOSuite
+import weaver.scalacheck.Checkers
+
+object SnapshotEventsPublisherDaemonSuite extends MutableIOSuite with Checkers {
+
+  override type Res =
+    (Supervisor[IO], Hasher[IO], SecurityProvider[IO], Metrics[IO], Encoder[CurrencySnapshotEvent], KeyPair)
```

Comment:
does `Encoder` for `CurrencySnapshotEvent` need to be a resource? it's a simple value, not a resource